./mach run --browserhtml on Windows

adamncasey opened the pull request:
This adds some windows specific behaviour to the ./mach run --browserhtml command:

Remove the webrender option
Convert the browserhtml path into a relative path to avoid a mingw->windows path conversion.

@larsbergstrom said that webrender support will come to windows ( https://github.com/servo/servo/issues/10243#issuecomment-215968632 ), which will remove the need for part of this soon. But it makes testing slightly nicer.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/10943)

<!-- Reviewable:end -->

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/servo/command_base.py, python/servo/post_build_commands.py

adamncasey commented:
I'm not totally convinced by converting it to a relative path. But it happens to work around the problem of msys python generating a unix format path, and servo expecting paths in windows' format.

larsbergstrom commented:
@bors-servo r+

I'm approving this so that it can be tested, but I do not expect browser.html to work at all well with the default CPU renderer. Like most other browser engines, it does quite poorly on such a complicated page.

bors-servo commented:
:pushpin: Commit 738ae67 has been approved by `larsbergstrom`

<!-- @bors-servo r=larsbergstrom 738ae67cec53a99fa136619578a7891c2d71227a -->

bors-servo commented:
:hourglass: Testing commit 738ae67 with merge f935a14...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/666)

adamncasey commented:
I'll fix the tidy problems & push a new commit soon

UK992 commented:
With https://github.com/servo/webrender/pull/247 made Webrender actually work well on Windows. I've been testing it for a month now.

UK992 commented:
Now that Webrender PR for Windows was merged, may i suggest to exclude command ``-b`` instead, because window without native titlebar is currently useless on Windows.

adamncasey commented:
That seems sensible

adamncasey commented:
Does disabling the titlebar work on Linux? It might make more sense to only pass -b if it's running on a mac

paulrouget commented:
> Does disabling the titlebar work on Linux?

No.

adamncasey committed:
Only enable borderless browserhtml window on OSX.

Pass browserhtml path as relative path to avoid a mingw->windows path conversion

highfive commented:
New code was committed to pull request.

adamncasey commented:
I don't have an appropriate box to test this on, but borderless should now only enable on OSX.

When the dependency on webrender is updated past https://github.com/servo/webrender/commit/672a36000e09f05f579569b8dc9330152fa984b8 --browserhtml will also work on Windows

Jayflux commented:
@larsbergstrom ^

bors-servo commented:
:umbrella: The latest upstream changes (presumably #11332) made this pull request unmergeable. Please resolve the merge conflicts.

metajack commented:
Windows webrender seems to work fine on master. What am I missing?

<details><summary><i>Previously, bors-servo wrote…</i></summary>
> :umbrella: The latest upstream changes (presumably #11332) made this pull request unmergeable. Please resolve the merge conflicts.
</details>


---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[python/servo/command_base.py, line 110 \[r1\]](https://reviewable.io:443/reviews/servo/servo/10943#-KIcbWP06PY6VH6QfaKE:-KIcbWP06PY6VH6QfaKF:-1572445321) ([raw file](https://github.com/servo/servo/blob/b3b0a20c34396e5bc225319a338fc17e410af4ee/python/servo/command_base.py#L110)):*
> ```Python
>     if sys.platform == 'win32':
>         return True
>     if platform.system() == 'Windows':
> ```

I think this can be written simply as:

return sys.platform == 'win32' or sys.platform = 'msys'

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10943#-:-KIctnVg_lm59toQqsDW:1999727232)*
<!-- Sent from Reviewable.io -->

larsbergstrom commented:
I don't think we need this anymore, now that WebRender on master works on Windows. Hooray!

UK992 commented:
@larsbergstrom Webrender is solved, but latest commit which disable useless borderless window and fix browser.html path related issue, which show only blank page.
